src/Application.php

```diff
@@ -25,6 +29,48 @@ public function getTransformationConfig(int $transformationId): array
         return $this->removeDisableTransformation($transformationConfig);
     }
 
+    public function migrateTransformationConfig(array $transformationConfig): array
+    {
+        $transformationsV2 = [];
+        foreach ($transformationConfig['rows'] as $row) {
+            $transformationKey = sprintf(
+                '%s-%s',
+                $row['configuration']['backend'],
+                $row['configuration']['type']
+            );
+
+            $transformationV2 = new TransformationV2(
+                $transformationConfig['name'],
+                $row['configuration']['type'],
+                $row['configuration']['backend']
+            );
+            if (isset($transformationsV2[$transformationKey])) {
```

Comment:
ok, uz som na to prisiel, ze som prehliadol tento riadok. ... to vytvaranie `new TransformationV2(` by som dal do `else` aby to bolo lepsie pochopitelne ako clovek cita kod odvrchu -> existuje? `pouzijeme existujucu`, inak `vytvor objekt`

Riadok na konci foreach `$transformationsV2[$transformationKey] = $transformationV2;` ma napr. zmysel iba pri vytvarani.

Ako je to funkcne, ale fakt som nebol schopny na prvy pohlad prist na to ako to funguje.

Takto by to bolo pochopitelnejsie:
```
        $transformationsV2 = [];
        foreach ($transformationConfig['rows'] as $row) {
            $transformationKey = sprintf(
                '%s-%s',
                $row['configuration']['backend'],
                $row['configuration']['type']
            );

            if (isset($transformationsV2[$transformationKey])) {
                $transformationV2 = $transformationsV2[$transformationKey];
            } else {
                $transformationV2 = new TransformationV2(
                    $transformationConfig['name'],
                    $row['configuration']['type'],
                    $row['configuration']['backend']
                );
                $transformationsV2[$transformationKey] = $transformationV2;
            }

            $this->processRow($row, $transformationV2);
        }
```

`processRow` moze byt void, kedze ten isty objekt vlezie do metody aj z nej vyjde.

Tak zvaz ci ti to dava zmysel upravit alebo nie. Idem to pozriet cele.

Reply:
ok, upravil jsem to tak by to mělo být čitelnější